File: BuyFoodServer/src/main/java/org/duckdns/mancitiss/buyfoodserver/Product.java
package org.duckdns.mancitiss.buyfoodserver;

public class Product {
    public String name;
    public long price;
    public String category;
    public String id;
    public int quantity;
    public String description;
    public long created;
    public long stars;
    public long ratingCount;

    public String avatar;

    public Product(){
    }

    // initialize
    public Product(String id, String name, long price, String category, int quantity, String description, long created, long stars, long ratingCount) {
        this.id = id;
        this.name = name;
        this.price = price;
        this.category = category;
        this.quantity = quantity;
        this.description = description;
        this.created = created;
        this.stars = stars;
        this.ratingCount = ratingCount;
    }

    public Product(String id, String name, long price, String category, int quantity, String description, long created, long stars, long ratingCount, String avatar) {
        this.id = id;
        this.name = name;
        this.price = price;
        this.category = category;
        this.quantity = quantity;
        this.description = description;
        this.created = created;
        this.avatar = avatar;
        this.stars = stars;
        this.ratingCount = ratingCount;
    }

}

File: TestApplication/app/src/main/java/org/duckdns/mancitiss/testapplication/Product.kt
package org.duckdns.mancitiss.testapplication

import android.graphics.Bitmap
import java.io.Serializable

class Product () : Serializable{
    var name: String = ""
    var price: Long = 0
    var category: String = ""
    var id: String = ""
    var quantity = 0
    var description: String = ""
    var created: Long = 0
    var stars: Long = 0
    var ratingCount: Long = 0

    var averageStars: Float = 0f

    var avatar: String = ""
    var image: Bitmap? = null

    //constructor() : this() {}

    constructor(
        id: String,
        name: String,
        category: String,
        price: Long,
        quantity: Int
    ) : this() {
        this.id = id
        this.name = name
        this.category = category
        this.price = price
        this.quantity = quantity
    }
}